[added] Mounted Bow to vehicle shop + Mounted Bow changes

mugg91 opened the pull request:
## Status

- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

[added] Mounted Bow to vehicle shop, can be bought for 80 coins
[changed] Mounted Bow can be picked up unless it was loaded with the map
[changed] Mounted Bow health from 2.0 to 2.5
[changed] Mounted Bow properties, making it heavier
[changed] Mounted Bow width is just wide enough to not fit in 1 tile gaps
[changed] Mounted Bow inventory space from 3 slots to 2 slots
[changed] Mounted Bow arrows won't despawn early
[changed] Mounted Bow shoot delay from 25 ticks to 35 ticks
[changed] Mounted Bow will not fall through platforms and doors

Fixes https://github.com/transhumandesign/kag-base/issues/1741
Fixes https://github.com/transhumandesign/kag-base/issues/2205
Fixes https://github.com/transhumandesign/kag-base/issues/2203
Fixes https://github.com/transhumandesign/kag-base/issues/2204

I added Mounted Bow to vehicle shop so it can be used in CTF which was sometimes requested by players, to add a bit more variety to the game.

I made some changes to Mounted Bow:
* It was a featherweight item and could be thrown far across the map. I changed that to make it heavier.
* I gave it more health so it isn't destroyed and 80 coins were wasted right away
* I made it not fit in 1 tile gaps because players could attach themselves to Mounted Bow and fall down through gaps.
* I also changed the shape vertically so it looks like it actually touches the ground (was floating 1 px high in the air previously)
* I gave it less inventory space in an effort to make it less of a camping tool
* I made it so arrows don't despawn as I like it better that way. I don't think making it a short distance tool in CTF is a good idea.

----

~My only concern now is maybe we should make it shoot at a lesser frequency.
It fires slightly too often imo.
What do you think?~

## Steps to Test or Reproduce

Go to Sandbox or CTF.
Spawn vehicle shop, 
purchase a Mounted Bow and play around with it.

mugg91 committed:
[added] Mounted Bow to vehicle shop + changes

mugg91 commented:
Should Mounted Bow be able to convert to the other team? By pickup or by waiting at it?

mugg91 committed:
small fixes

mugg91 commented:
I made a commit that does the following changes:

I changed how Mounted Bow behaves as a medium weight item.
* Tag "medium weight" was added to Mounted Bow which actually does what I had intended, reducing throw distance
* mass reduced from 150 to 100
* friction changed from 0.5 to 0.6
* elasticity changed from 0.1 to 0.3
* buoyancy changed from 0.1 to 0.2
* drag changed from 1.0 to 0.5

I addressed the issue with how it shoots too frequently; it could be too spammy in CTF
* shoot delay changed from 25 ticks to 35 ticks

I made mounted bow possible to steal from enemy team by pick-up
* Added `SetTeamToCarrier.as` as blob script to `MountedBow.cfg`

GELGELAT commented:
its really cool. You have to look at it relative to regular archers. I dont think hes supposed to shoot faster than them. Imagine if 3 people shoot it like a turret. maybe make it a team item?

mugg91 commented:
What do you mean by making it a team item?
Right now, it can only be mounted by team players but can be stolen by the other team via pickup.
Should I add a check like in Saw, that it can only be picked from below or when overlapping?

GELGELAT commented:
> What do you mean by making it a team item? Right now, it can only be mounted by team players but can be stolen by the other team via pickup. Should I add a check like in Saw, that it can only be picked from below or when overlapping?

let's go like a saw, either way its worth a lot of coin to just steal.

mugg91 commented:
I think we should add a wood cost, too. 100 or 150 wood. Because there is usually plenty wood anyway and buying mounted bow shouldn't be something someone can just spam.

> let's go like a saw, either way its worth a lot of coin to just steal.

Of course, bringing it to the battle field is risky.
I've though about it more like fortifying your castle and placing it behind platforms so it can't be picked  (assuming this will be in the next build, too: https://github.com/transhumandesign/kag-base/pull/1864 ). 

I think it's not too far off compared to if you mess up your Keg or 2-3 bombs. Or when you think about bringing Saw to the battle field, that's also a high risk for wasting 100 stone, so Saw is usually used only for industry.

GELGELAT commented:
> Should we add a wood cost? Like 100 or 150 wood. Because there is usually plenty wood anyway and buying mounted bow shouldn't be something someone can just spam.
> 
> > let's go like a saw, either way its worth a lot of coin to just steal.
> 
> Of course, bringing it to the battle field is risky. I was thinking about it more like fortifying your castle and placing it behind platforms so it can't be picked (assuming this will be in the next build, too: #1864 ).
> 
> I think it's not too far off compared to if you mess up your Keg or 2-3 bombs. Or when you think about bringing Saw to the battle field, that's also a high risk for wasting 100 stone, so Saw is usually used only for industry.


The cost in the form of wood is not so necessary this resource can be obtained everywhere, I think it will only be a distraction. Hmmm.

GELGELAT commented:
I have an idea. What if you let the bow be picked up when theres already a person sitting in it? What do you think?

GELGELAT commented:
you could do that with 2 or more units. Sounds creepy, but like for fun

mugg91 committed:
fixes

Buying from Vehicleshop won't cause an error in the console.
Made Mounted Bow to collide with platforms and doors.
Mounted Bow can only be picked up from afar by same team, or from enemies when overlapping.

mugg91 committed:
Merge branch 'master' into mugg91-mounted-bow-purchasable

mugg91 commented:
I committed changes:
* Somehow `VehicleShop.as` had file conflicts that are now resolved - Buying from vehicle shop won't cause errors in the console now.
* Mounted Bow will not fall through doors and platforms.
* Mounted Bow can be picked by allies from afar, and from enemies only when overlapping (same as crates).

mugg91 committed:
forgotten file

CollidesOnlyWithLarger needs to removed to allow collision with doors and platforms

mugg91 committed:
Merge branch 'mugg91-mounted-bow-purchasable' of https://github.com/mugg91/kag-base into mugg91-mounted-bow-purchasable
